network/config/netconfig.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package config
+
+// TAPConfig defines a TAP interface configuration.
+type TAPConfig struct {
+	Uid    int
+	Gid    int
+	Queues int
```

Comment:
Those do not require json tags as I understand. Because they are not marshaled/un-marshaled to/from json. The `netConfigJSON` struct in the plugins package is used for that.

Reply:
Do we think there is value in reusing a struct like this? IMO this is micro-sharing, creates unnecessary coupling dependency without any real value.

Reply:
I don't think the shared "TAPConfig" abstraction provides any value here. If we are thinking seriously about reusing tap TAP related logic between plugins, we could create a new "network/tap" package that defines a new class with methods of creating and deleting TAP interfaces.